src/GraphQL/Types/Composite/AutoRegisteringObjectGraphType.cs

```diff
@@ -75,78 +101,102 @@ internal static class AutoRegisteringHelper
                     deprecationReason: propertyInfo.ObsoleteMessage()
                 ).DefaultValue = (propertyInfo.GetCustomAttributes(typeof(DefaultValueAttribute), false).FirstOrDefault() as DefaultValueAttribute)?.Value;
             }
         }
 
-        private static bool IsNullableProperty(PropertyInfo propertyInfo)
+        internal static void SetFields<TSourceType>(ComplexGraphType<TSourceType> type, IEnumerable<MethodInfo> methods, params Expression<Func<TSourceType, object>>[] excludedMethods)
         {
-            if (Attribute.IsDefined(propertyInfo, typeof(RequiredAttribute))) return false;
+            foreach (var methodInfo in methods)
+            {
+                if (excludedMethods?.Any(p => GetMemberInfo(p) == methodInfo) == true)
+                    continue;
 
-            if (!propertyInfo.PropertyType.IsValueType) return true;
+                var methodResolver = new MethodResolver(methodInfo);
 
-            return propertyInfo.PropertyType.IsGenericType && propertyInfo.PropertyType.GetGenericTypeDefinition() == typeof(Nullable<>);
+                methodResolver.CreateQueryArguments();
+
+                var result = type.Field(
+                    type: methodInfo.ReturnType.GetGraphTypeFromType(methodInfo.GraphTypeIsNullable()),
+                    name: methodInfo.Name,
+                    arguments: methodResolver.CreateQueryArguments(),
+                    description: methodInfo.Description(),
+                    deprecationReason: methodInfo.ObsoleteMessage()
+                );
+
+                result.Resolver = methodResolver;
+            }
         }
 
-        private static string GetPropertyName<TSourceType>(Expression<Func<TSourceType, object>> expression)
+
+        private static MemberInfo GetMemberInfo(LambdaExpression expression)
         {
-            if (expression.Body is MemberExpression m1)
-                return m1.Member.Name;
+            var expr = (expression.Body is UnaryExpression u) ? u.Operand : expression.Body;
+
+            if (expr is MemberExpression m)
+                return m.Member;
 
-            if (expression.Body is UnaryExpression u && u.Operand is MemberExpression m2)
-                return m2.Member.Name;
+            if (expr is MethodCallExpression c)
+                return c.Method;
 
-            throw new NotSupportedException($"Unsupported type of expression: {expression.GetType().Name}");
+            throw new NotSupportedException($"Unsupported type of expression: {expr.GetType().Name}");
         }
 
-        internal static bool IsEnabledForRegister(Type propertyType, bool firstCall)
+        internal static bool IsEnabledForRegister(Type returnType, bool firstCall)
         {
-            if (propertyType == typeof(string)) return true;
+            if (returnType == typeof(string))
+                return true;
 
-            if (propertyType.IsValueType) return true; // TODO: requires discussion: Nullable<T>, enums, any struct
```

Comment:
Please revert all code formatting changes. These changes are not relevant and only complicate the review.

Reply:
done